v21.2/deploy-lambda-function.md

```diff
@@ -0,0 +1,164 @@
+---
+
+This tutorial shows you how to create an [AWS Lambda](https://aws.amazon.com/lambda) function that communicates with a {{ site.data.products.serverless }} cluster.
+
+The example function used for this tutorial is written in Python. The function uses the [Psycopg](https://www.psycopg.org/) PostgreSQL adapter to connect to CockroachDB.
+
+## Prerequisites
+
+Before starting the tutorial, do the following:
+
+1. Create a [{{ site.data.products.db }}](https://cockroachlabs.cloud/signup?referralId={{page.referral_id}}) account.
+
+1. Create an [AWS](https://aws.amazon.com/) account.
+
+1. Install the [AWS CLI](https://aws.amazon.com/cli/).
+
+## Step 1. Create a {{ site.data.products.serverless }} cluster
+
+{% include cockroachcloud/quickstart/create-a-free-cluster.md %}
+
+<a name="connection-string"></a>
+
+After the cluster is created, the **Connection info** window appears. Click the **Connection string** tab and copy the connection string to a secure location. You will use this connection string to connect to CockroachDB later in the tutorial.
+
+{{site.data.alerts.callout_info}}
+The connection string is pre-populated with your username, cluster name, and other details, including your password. Your password, in particular, will be provided only once. Save it in a secure place (we recommend a password manager) to connect to your cluster in the future. If you forget your password, you can reset it by going to the [**SQL Users** page](../cockroachcloud/user-authorization.html).
+{{site.data.alerts.end}}
+
+## Step 2. Get the sample code
+
+Open a terminal window and copy the sample code's GitHub repo:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/examples-aws-lambda/
+~~~
+
+The function's code is available under the `examples-aws-lambda/python` directory:
+
+~~~ shell
+.
+├── README.md
+├── deployment-package.zip  ## Lambda deployment package
+├── init_db.py              ## Lambda function source code
+├── package                 ## Psycopg dependencies
+├── requirements.txt        ## List of Python requirements
+└── root.crt                ## CA cert
+~~~
+
+## Step 3. (Optional) Create the deployment package
+
+{{site.data.alerts.callout_info}}    
+This step is optional, as you do not need to create a new deployment package to deploy the sample function. The `examples-aws-lambda` repo includes a deployment package that is ready to deploy.
+{{site.data.alerts.end}}
+
+1. Download and install the `psycopg2-binary` Python library to a new directory:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ python3 -m pip install --only-binary :all: --platform manylinux1_x86_64  --target ./my-package -r requirements.txt
+    ~~~
+
+    {{site.data.alerts.callout_info}}    
+    To run on Amazon Linux distributions, `pscyopg2` dependencies must be compiled for Linux.
+    {{site.data.alerts.end}}
+
+1. Compress the project files to a ZIP file for deployment:
```

Comment:
A `deployment-package.zip` file is checked into the code repo in cockroachlabs/examples-aws-lambda#1. This step creates a different ZIP file. Are both necessary?

Reply:
For the purposes of completing the tutorial, no. Two deployment packages are not necessary. Because of that, I've marked this step as optional in two places. 

Something to note is that it's likely that users will end up creating their own functions (or modifying the sample function we provide). If they want to deploy their own code, they'll need to know how to package the code with the right dependencies. So I think it's important to include instructions on how to manually package the ZIP file.